client/mysqlbinlog.cc

```diff
@@ -744,9 +748,12 @@ static void convert_path_to_forward_slashes(char *fname)
 */
 static bool shall_skip_database(const char *log_dbname)
 {
-  return one_database &&
-         (log_dbname != NULL) &&
-         strcmp(log_dbname, database);
+  if(log_dbname == NULL)
+    return false;
+
+  if (one_database)
+    return filter_databases.count(log_dbname) == 0;
```

Comment:
As far as I know `std::set`'s `.count()` overload that would take a `const char*` in this situation is only available starting with c++14. Pre c++14, `.count()` only takes `Key` type which is `std::string` in this case. I don't know if c++14 is available on all platforms for the server codebase. @andrelkin  can you help me out here?

Reply:
Hi @andrelkin ,I need your help.

Reply:
Hi! Sorry for delay (I thought there's some amount of review work), to the question
https://en.cppreference.com/w/cpp/container/set/count
specifies as c++14 **only** the **template** version of the method: 

`template< class K >  size_type count( const K& x ) const; | (2) | (since C++14)`